### DISCUSSION TITLE
Feature- M117 handling

### DISCUSSION THREAD

**Discussion by Alex00101** [KlipperScreen/KlipperScreen#337] [Ideas]
Apologies if this is not the right place for or way of making feature requests or if this has been discussed before.

Would it be possible to make KlipperScreen to handle M117 (message) command - e.g. to display even while printing in a popup window or pull down bat as it does now for errors?

Background - my printer initially came with PanelDue. I have added KlipperScreen, which is a much better solution, and want to retire PanelDue. The only thing that is outstanding is that my macros do use M177 messages a lot (e.g. "Unload complete. Remove filament") and while PanelDue is showing them in a pop up window, KlipperScreen currently does not. 

I'm using workaround now (RESPOND command of Error type) but not sure if it is the "right" way hence asking for M117 handling.

Thanks

**Comment by jordanruthe** [comment]
c8420c260bec7474637f79a77649b9a414d5a171 will now show a popup on a RESPOND command with type echo. M117 will show up in the job status window under the GCode file name. The error type of respond will show a different color banner.

**Comment by Alex00101** [comment]
Thanks, that was super fast!
